dfsclient/src/main/java/org/hpccsystems/dfs/client/BinaryRecordReader.java

```diff
@@ -868,43 +875,31 @@ private String getNullTerminatedString(HpccSrcType stype) throws IOException
                 }
                 else
                 {
-                    scratchStrBuilder.append(new String(scratchBuffer, 0, readSize, charset));
+                    strByteLen += readSize;
                 }
             }
         }
         else
         {
-            int remainingData = this.inputStream.available();
-            for (int i = 0; i < remainingData - 1;)
+            int eosLocation = -1;
```

Comment:
we could move the eosLocation declaration and initialization outside of the the if and the else block (line 847)